pycbc/inference/models/gated_gaussian_noise.py

```diff
@@ -207,24 +316,25 @@ def get_gate_times_hmeco(self):
         dict :
             Dictionary of detector names -> (gate start, gate width)
         """
-        params = self.current_params
-        # gate input for ringdown analysis which consideres a start time
-        # and an end time
-        dgate = params['gate_window']
         # generate the template waveform
         try:
-            wfs = self.waveform_generator.generate(**params)
+            wfs = self.get_waveforms()
         except NoWaveformError:
             return self._nowaveform_logl()
         except FailedWaveformError as e:
             if self.ignore_failed_waveforms:
                 return self._nowaveform_logl()
             raise e
-
+        # gate input for ringdown analysis which consideres a start time
+        # and an end time
+        dgate = params['gate_window']
```

Comment:
This is definitely a bug, no? Youa re reading from params before it is generated on the next line. I suspect this hasn't been run in exactly this state. Perhaps some patches missing? @cdcapano

Reply:
Yeah, that's a bug. I switched the order of the lines to fix it. We haven't actually been using this function, which is why I didn't catch it.